src/main/java/rs/teslaris/core/aspect/PublicationEditCheckAspect.java

```diff
@@ -0,0 +1,121 @@
+
+        String role = tokenUtil.extractUserRoleFromToken(tokenValue);
+        int personId = userService.getPersonIdForUser(tokenUtil.extractUserIdFromToken(tokenValue));
+
+        List<Integer> contributors = getContributors(annotation, attributeMap, joinPoint);
+
+        checkPermission(role, personId, contributors);
+
+        return joinPoint.proceed();
+    }
+
+    private Method getMethod(ProceedingJoinPoint joinPoint) {
+        var signature = joinPoint.getSignature();
+        if (!(signature instanceof MethodSignature)) {
+            throw new IllegalArgumentException("Aspect should be applied to a method.");
+        }
+        return ((MethodSignature) signature).getMethod();
+    }
+
+    private List<Integer> getContributors(PublicationEditCheck annotation,
+                                          Map<String, String> attributeMap,
+                                          ProceedingJoinPoint joinPoint) {
+        if (annotation.value().equalsIgnoreCase("CREATE")) {
+            return getContributorsFromDTO(joinPoint);
+        } else {
+            return getContributorsFromDatabase(attributeMap);
+        }
+    }
+
+    private List<Integer> getContributorsFromDTO(ProceedingJoinPoint joinPoint) {
+        JournalPublicationDTO publicationDTO = (JournalPublicationDTO) joinPoint.getArgs()[0];
+        return publicationDTO.getContributions().stream().map(PersonContributionDTO::getPersonId)
+            .filter(Objects::nonNull).collect(Collectors.toList());
+    }
+
+    private List<Integer> getContributorsFromDatabase(Map<String, String> attributeMap) {
+        int publicationId = Integer.parseInt(attributeMap.get("publicationId"));
+        return documentPublicationService.readJournalPublicationById(publicationId)
+            .getContributions().stream().map(PersonContributionDTO::getPersonId)
+            .filter(Objects::nonNull).collect(Collectors.toList());
+    }
+
+    private void checkPermission(String role, int userId, List<Integer> contributors) {
+        UserRole userRole = UserRole.valueOf(role);
+        switch (userRole) {
+            case ADMIN:
+                break;
+            case RESEARCHER:
+                contributors.forEach(System.out::println);
+                System.out.println(userId);
+                if (!contributors.contains(userId)) {
+                    throw new CantEditPublicationException(
+                        "You do not have the permission to edit this publication.");
+                }
+                break;
+            case INSTITUTIONAL_EDITOR:
+                if (contributors.stream().noneMatch(
+                    personId -> personService.isPersonEmployedInOrganisationUnit(personId,
+                        userService.getUserOrganisationUnitId(userId)))) {
+                    throw new CantEditPublicationException(
+                        "You do not have the permission to edit this publication.");
+                }
+                break;
```

Comment:
are you always using break in the last option of case? if yes, you can keep this line